lib/rules/web/admin_console/4.4/monitoring_metrics.xyaml

```diff
@@ -0,0 +1,139 @@
+click_primary_menu_developer_mode:
+  element:
+    selector:
+      xpath: //a[text()='<menu_text>']
+    op: click
+goto_monitoring_developer_mode:
+  params:
+    menu_text: Monitoring 
+  action: click_primary_menu_developer_mode
+click_projects_dropdown_developer_mode:
+  element:
+    selector:
+      xpath: (//div[@class='co-namespace-bar']//button)[1]
+    op: click
+    timeout: 3600
+input_select_project_developer_mode:
+  element:
+    selector:
+      xpath: //div[@class='co-namespace-selector']//input[@placeholder='Select project...']
+    op: send_keys <project_name>
+    type: input
+click_project_link_developer_mode:
+  element:
+    selector:
+      xpath: //a[text()='<project_name>']
+    op: click
+choose_project_developer_mode:
+  action: click_projects_dropdown_developer_mode
+  action: input_select_project_developer_mode
+  action: click_project_link_developer_mode
+check_tech_preview_badge:
+  element:
+    selector:
+      xpath: //span[text()='Tech Preview']
+click_tab_developer_mode:
+  element:
+    selector:
+      xpath: //a[text()='<tab_name>']
+    op: click
+
+perform_cutomer_query:
+  action: click_metrics_query_dropdown
+  action: click_custom_query_button
+  action: check_query_input_text_area
+  action: set_query_input_text_area
+
+click_metrics_query_dropdown:
+  element:
+    selector:
+      xpath: //div[@class='co-m-pane__body' ]//div[contains(@class, 'odc-metrics-query-input')]//button
+    op: click
+
+click_custom_query_button:
+  element:
+    selector:
+      xpath: //button[text()='Custom Query']
+    op: click
+set_query_input_text_area:
+  elements:
+  - selector: 
+      xpath: //div[@class='co-m-pane__body']//textarea
+    op: send_keys <metrics>
+    type: textarea
+  - selector: 
+      xpath: //div[@class='co-m-pane__body']//textarea
+    op: send_keys <press_enter>
+    type: textarea 
+
+choose_metrics_query:
+  element:
+    selector:
+      xpath: //a[text()='<metrics_name>']
+    op: click
+check_query_input_text_area:
+  element:
+    selector: 
+      xpath: //div[@class='co-m-pane__body']//textarea
+click_promql_button:
```

Comment:
can be replaced with `click_button_text`

Reply:
Changed the code here, but I continue to define the xpath here, because I defined anchor here and used many times.